module/window_swallowing.lua

```diff
@@ -56,6 +56,8 @@ local function manage_clientspawn(c)
     local parent_client = awful.client.focus.history.get(c.screen, 1)
     if not parent_client then
         return
+    elseif parent_client.type == "dialog" or parent_client.type == "splash" then
+        return
```

Comment:
:+1:

Reply:
as mentioned earlier some programs with splash windows can have the splash window as parent process of the main window, and in the case of krita it's splash wrongfully has the dialog type, now if krita is the only program that has that bug i don't know, will have to keep checking the window type of programs with splash windows to be sure.